code/src/components/ClearList.js

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable max-len */
+import React, { useState, useRef, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import tickets from 'reducers/tickets';
+import { ClearButtons } from './buttons';
+
+const ClearList = () => {
+  const dispatch = useDispatch();
+  const [clickCount, setClickCount] = useState(0);
+  const buttonRef = useRef(null);
+
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (buttonRef.current && !buttonRef.current.contains(event.target)) {
+        setClickCount(0);
+      }
+    };
+    window.addEventListener('click', handleClickOutside);
+    return () => {
+      window.removeEventListener('click', handleClickOutside);
+    };
+  }, [buttonRef]);
+
+  const handleClearAllClick = () => {
+    if (clickCount === 0) {
+      setClickCount(1);
+    } else {
+      dispatch(tickets.actions.clearAll());
+      setClickCount(0);
+    }
+  };
+
+  const handleAllDoneClick = () => {
```

Comment:
Clever way of handling when all tasks are done! I will def remember that for future projects.